### DISCUSSION TITLE
Row Level Security for Different Types of Users?

### DISCUSSION THREAD

**Discussion by osseonews** [supabase/supabase#4405] [Questions]
Our blog app has a concept of authors and editors. Authors can only see their posts, while editors can see their owns posts, as well as all any authors post. As this app runs in the client we can't use the service role key. So how do you create policies for something like this with RLS?  Every sample we've seen for RLS is an all or nothing approach, i.e. authenticated users can see all of their own posts, and none of the others. But what about, an authenticated user who can see their own posts, as well as everyone elses, because they are an editor? Do we need to create 2 separate rules like below to accommodate various types of users? Will editors 
be able to see all posts, even if the first row policy fails?

create policy "Authors can view their own posts." 
    on posts for select 
    using ( auth.uid() = user_id );

create policy "Editor can view all posts." 
    on posts for select 
    using ( auth.uid() in (
      select user_id from users
      where role = "editor"
    ) )

**Comment by kk21** [comment]
You can create multiple policies since policies are permissive by default (combined using boolean OR):
https://www.postgresql.org/docs/13/sql-createpolicy.html
